### PR TITLE
Adds ability to customize gradient

### PR DESCRIPTION
**Pull request by vinnyjth**
### What does this PR do, or why is it needed?

Adds a section in the theme where you can customize the "overlays" used by LinearGradient for the following components

- BackgroundView
- GradientOverlayImage

For BackgroundView, you can customize the overlay by overriding the existing `background-gradient` overlay. For the GradientOverlayImage, you can now supply a `overlayType` that matches one of the options inside the theme. 

### What design trade-offs/decisions were made?

Might need a better API for consistent overlays. Right now it's not consistent what `args` are passed to the overlay function. 

### How do I test this PR?

Check out the stories for BackgroundView and GradientOverlayImage

### What automated tests did you write?

Updated snaps. 

## TODO:

- [x] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [x] References #851 
- [x] No new warnings in tests, in storybook, and in-app
- [x] Upload GIF(s) of iOS and Android
- [x] Set a relevant reviewer

## REVIEW:

**Manual QA**

- [x] Manual QA on iOS and ensure it looks/behaves as expected
- [x] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [x] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [x] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [x] Could the code be simpler?
- [x] Will the code be easy to modify in the future?
- [x] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

**Comment by vinnyjth** [comment]
Simulator Screen Shot - iPhone X - 2019-07-16 at 08 40 38

**Comment by burkeshartsis** [comment]
it looks like after merging master it some schema stuffs broke

File: packages/apollos-ui-kit/src/theme/__snapshots__/createTheme.tests.js.snap [hunk 1]
```diff
@@ -82,6 +82,16 @@ Object {
     "rem": [Function],
     "verticalRhythm": [Function],
   },
+  "overlays": Object {
+    "background-gradient": [Function],
```
[the remark this answers is not in the document]
Comment (vinnyjth) [reply packages/apollos-ui-kit/src/theme/__snapshots__/createTheme.tests.js.snap]: I actually didn't write these custom snapshots, they just ended up in the `theme` snapshots by default :)

File: packages/apollos-ui-kit/src/GradientOverlayImage/index.js [hunk 2]
```diff
@@ -3,28 +3,17 @@ import { StyleSheet, View } from 'react-native';
...
-      overlayColor,
-    ],
-    start: { x: 0, y: 0 },
-    end: { x: 0, y: 1 },
-    locations: [0.5, 1],
-  };
-
-  return values;
-};
+const ThemedOverlay = withTheme(
```
[the remark this answers is not in the document]
Comment (vinnyjth) [reply packages/apollos-ui-kit/src/GradientOverlayImage/index.js]: ❤️

**Pull request merged**